spmd/tensor/ops/matrix_ops.py

```diff
@@ -1,27 +1,60 @@
 # Copyright (c) Meta Platforms, Inc. and affiliates
 # implement matrix related ops for distributed tensor
-from typing import Optional
-from spmd.tensor.dispatch import OpSchema
-from spmd.tensor.placement_types import PlacementSpec
+from spmd.tensor.dispatch import OpSchema, OutputSharding
 from spmd.tensor.ops.prop_rules import einop_prop, mm_prop, pointwise_prop
+from spmd.tensor.placement_types import _Partial, PlacementSpec, Shard
 from spmd.tensor.ops.utils import register_prop_rule
 
 
 @register_prop_rule("aten.mm.default")
-def mm_rules(op_schema: OpSchema) -> Optional[PlacementSpec]:
+def mm_rules(op_schema: OpSchema) -> OutputSharding:
     mat1_spec, mat2_spec = op_schema.args_spec
-    return mm_prop(mat1_spec, mat2_spec)
+    return OutputSharding(mm_prop(mat1_spec, mat2_spec))
 
 
 @register_prop_rule("aten.addmm.default")
-def addmm_rules(op_schema: OpSchema) -> Optional[PlacementSpec]:
+def addmm_rules(op_schema: OpSchema) -> OutputSharding:
     input_spec, mat1_spec, mat2_spec = op_schema.args_spec
     mm_out_spec = mm_prop(mat1_spec, mat2_spec)
     if mm_out_spec is None:
-        return None
-    return pointwise_prop((input_spec, mm_out_spec))
+        # non-eligible input, suggest addmm input specs
```

Comment:
Maybe you can do instead something like?


  return pointwise_rules([mm_out_spec.placements[0], input_spec], linear=True)

Reply:
could you elaborate more on what this mean? here we are trying to return a suggestion that contains a OpSchema of args/kwargs, if we return a pointwise_rules with `linear=True`, wouldn't this just propagate through without actually doing the reshard? or are you saying that we can directly run this, and don't set the output_spec of the output sharding, but rather setting the `op_schema` passed in to the `schema_suggestion` list?

Reply:
oh i put my comment in the wrong line , see comment in the line below.